enterprise/micronaut/src/org/netbeans/modules/micronaut/hyperlink/HyperlinkImpl.java

```diff
@@ -0,0 +1,290 @@
+
+            if (title == null) {
+                title = p.get(titleKey, null);
+
+                if (title != null) {
+                    return title;
+                }
+            }
+
+            p.putLong(timestampKey,lastModified + (title != null ? TIME_VALID : TIME_INVALID));
+
+            if (title != null) {
+                p.put( titleKey, title);
+            } else {
+                p.remove(titleKey);
+            }
+
+            return title;
+        } catch (BadLocationException ex) {
+            Exceptions.printStackTrace(ex);
+        } catch (MalformedURLException ex) {
+            LOG.log(Level.FINE, null, ex);
+        }
+
+        return null;
+    }
+
+    private static final Pattern TITLE = Pattern.compile("<title>(.*)</title>");//NOI18N
+    private static final int MAX_NUM_BYTES_TO_READ = 10 * 1024;
+    private static String readTitle(URL url) {
+        ByteArrayOutputStream baos = null;
+        InputStream ins = null;
+        URLConnection c = null;
+        
+        try {
+            c = url.openConnection();
+
+            String encoding = null;
+            try {
+                encoding = c.getContentEncoding();
+            } catch (Throwable ex) {
+                return null;
+            }
+
+            if (encoding == null) {
+                encoding = Parser.decodeContentType(c.getContentType());
+            }
+
+            baos = new ByteArrayOutputStream();
+            ins = c.getInputStream();
+
+            int read;
+            
+            int numBytesRead = 0;
+            while ((read = ins.read()) != (-1) && numBytesRead < MAX_NUM_BYTES_TO_READ) {
+                baos.write(read);
+                numBytesRead++;
+            }
+
+            ins.close();
+            baos.close();
+
+            String content = new String(baos.toByteArray(), encoding != null ? encoding : Charset.defaultCharset().name());
+            Matcher m = TITLE.matcher(content);
+
+            if (m.find()) {
+                return m.group(1).trim();
+            } else {
+                return "";
+            }
+        } catch (IOException ex) {
+            LOG.log(Level.FINE, url != null ? url.toString() : null, ex);
+        } catch (IllegalArgumentException iae) {
+            // #198333
+            LOG.log(Level.FINE, url != null ? url.toString() : null, iae);
+        } finally {
+            if (baos != null) {
+                try {
+                    baos.close();
+                } catch (IOException ex) {
+                    Exceptions.printStackTrace(ex);
+                }
+            }
+
+            if (ins != null) {
+                try {
+                    ins.close();
+                } catch (IOException ex) {
+                    Exceptions.printStackTrace(ex);
+                }
+            }
+        }
+
+        return null;
+    }
+    
+    @MimeRegistration(mimeType = "text/x-java", service = HyperlinkLocationProvider.class)
```

Comment:
Try to use empty mime path to register this provider for all files.

Reply:
That didn't work, but hopefully we won't need this part at all.